routing/router.go

```diff
@@ -1,43 +1,72 @@
 package routing
 
 import (
+
 	"net/http"
 
 	"github.com/gorilla/mux"
 
 	"github.com/kiali/kiali/config"
+
+	"github.com/kiali/kiali/log"
 )
 
 // NewRouter creates the router with all API routes and the static files handler
 func NewRouter() *mux.Router {
 
-	router := mux.NewRouter().StrictSlash(true)
+	conf := config.Get()
+	webRoot := conf.Server.WebRoot
+	webRootWithSlash := webRoot + "/"
+
+	rootRouter := mux.NewRouter().StrictSlash(false)
+	appRouter := rootRouter;
+
+	if webRoot == "/" {
+		// appRouter.StrictSlash(true)
+	} else {
+		// Due to PathPrefix matching behavoir on sub-routers
+		// we need to explicitly redirect /foo -> /foo/	
+		// See https://github.com/gorilla/mux/issues/31
+		rootRouter.HandleFunc(webRoot, func(w http.ResponseWriter, r *http.Request) {
+			http.Redirect(w, r, webRootWithSlash, http.StatusFound)
+		})
+		appRouter = rootRouter.PathPrefix(conf.Server.WebRoot).Subrouter().StrictSlash(true)
+	}
 
 	// Build our API server routes and install them.
-	routes := NewRoutes()
-	for _, route := range routes.Routes {
+	apiRoutes := NewRoutes()
+	for _, route := range apiRoutes.Routes {
 		var handlerFunction http.Handler
 		if handlerFunction = route.HandlerFunc; route.Authenticated {
 			handlerFunction = config.AuthenticationHandler(handlerFunction)
 		}
-		router.
+		appRouter.
 			Methods(route.Method).
 			Path(route.Pattern).
 			Name(route.Name).
 			Handler(handlerFunction)
 	}
-	conf := config.Get()
+
 	// All client-side routes are prefixed with /console.
 	// They are forwarded to index.html and will be handled by react-router.
-	router.PathPrefix("/console").HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		http.ServeFile(w, r, conf.Server.StaticContentRootDirectory+"/index.html")
+	appRouter.PathPrefix("/console").HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		http.ServeFile(w, r, conf.Server.StaticContentRootDirectory + "/index.html")
 	})
 
 	// Build our static files routes by first creating the file server handler that will serve
 	// the webapp js files and other static content. Then tell the router about our fixed
 	// routes which pass all static file requests to the file handler.
-	fileServerHandler := http.FileServer(http.Dir(conf.Server.StaticContentRootDirectory))
-	router.PathPrefix("/").Handler(fileServerHandler)
+	staticFileServer := http.FileServer(http.Dir(conf.Server.StaticContentRootDirectory))
+	if webRoot != "/" {
+		staticFileServer = http.StripPrefix(webRootWithSlash, staticFileServer)
+	}
+	appRouter.PathPrefix("/").Handler(staticFileServer)
 
-	return router
+	return rootRouter
+}
+
+func ConsoleIndexHandler(w http.ResponseWriter, r *http.Request) {
+	log.Infof("/console/index.html")
+	conf := config.Get()
+	http.ServeFile(w, r, conf.Server.StaticContentRootDirectory+"/index.html")
```

Comment:
trivial - I suspect the missing black spaces that should surround the + sign doesn't follow gofmt formatting - I think you need spaces around +.